packages/ui/CategorySelector/index.tsx

```diff
@@ -12,42 +12,47 @@ import {
   SupportingLabelStyled,
 } from "./CategorySelectorStyled";
 import { CategoryIcon, Icon } from "ui";
-import categoryMap from "lib/category-map";
-import { CategoryMap } from "lib/types";
 import { CategoryType } from "ui/CategoryIcon";
-import { useHasScrollBar } from "lib/hooks/useHasScrollBar";
 
 type CategorySelectorProps = {
   onValueChange: (value: string) => void;
   label?: string;
   errors?: string[];
+  categoryMap: Object;
+  hasScrollbar?: boolean;
 };
 
 export const CategorySelector = ({
   errors,
   onValueChange,
   label,
+  categoryMap,
+  hasScrollbar,
 }: CategorySelectorProps) => {
-  const { hasScrollbar } = useHasScrollBar();
   const [isOpen, setIsOpen] = useState(false);
   const hasErrors = errors ? errors.length > 0 : false;
+  type CategoryMapType = typeof categoryMap;
 
-  const getCategoriesItems = (categoryMap: CategoryMap) => {
+  const getCategoriesItems = (categoryMap: CategoryMapType) => {
     const categories = Object.keys(categoryMap);
 
     return categories.map((category) => (
       <SelectItemStyled value={category} key={category}>
         <Select.ItemText>
           <CategoryWrapperStyled>
```

Comment:
```suggestion
  const getCategoriesItems = (categoryMap: CategoryMapType) => {
    const categoryEntries = Object.entries(categoryMap);

    return categoryEntries.map(([categoryKey, category]) => (
      <SelectItemStyled value={categoryKey} key={categoryKey}>
        <Select.ItemText>
          <CategoryWrapperStyled>
            {category && <CategoryIcon small category={category} />}
            {category?.name && (
              <CategoryNameStyled>{category?.name}</CategoryNameStyled>
            )}
```
We can get rid of type casting by using Object.entries instead of Object.keys and destruct the entry for the category key and value inside the map

Reply:
Done